### ISSUE TITLE
Events

### ISSUE THREAD

**Issue by victorbosh** [typesense/typesense-instantsearch-adapter#26]
Is there any way to catch the response from TS server in browser as variable or something? I can't find it in Instantsearch.js
I mean - any time the search return hits I need somehow to get those and lets say send some analytics or something else

**Comment by jasonbosco** [comment]
@victorbosh Have a look at the [`analytics`](https://www.algolia.com/doc/api-reference/widgets/analytics/js/) widgets in InstantSearch.js. 

Here's an example where I used it to send analytics to GA: https://github.com/typesense/showcase-songs-search/blob/39b3a402da0d0df4cda458daeb3909e030e3ba58/src/app.js#L187-L196

**Comment by victorbosh** [comment]
I tried it but it is deprecated. I also tried upgrade to insights middleware as mentioned here https://www.algolia.com/doc/guides/building-search-ui/upgrade-guides/js/#analytics-widget
But I can not even start to list the problems for example events are fired randomly times (just like 'render' event too) which makes no sense at all. I spent last week just dealing with Instantsearch.  It is more tricky to implement that stated. Well, thank you anyway
